web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -86,74 +86,99 @@ CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SOURCE_SCHEMA VARCHA
-                  }
-              }
-              add_columns_to(TARGET_SCHEMA, TARGET_TABLE, columnAdditions.join(','));
-              return "ok.Columns added."
-          }
-      } else {
-          return "ok. Columns match."
-      }
+  column_check_stmt = `
+    WITH source_data AS (
+      SELECT
+        column_name,
+        ordinal_position,
+        character_maximum_length,
+        data_type,
+        CASE
+          WHEN isc.data_type='TEXT' THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
+          WHEN isc.data_type='NUMBER' THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
+          else CONCAT(isc.column_name, ' ', isc.data_type)
+        END AS column_definition
+      FROM information_schema.columns isc
+      WHERE table_schema = UPPER(:1)
+      AND table_name = UPPER(:2)
+      ORDER BY ordinal_position
+      ),
+
+  target_data AS (
+    SELECT
+      column_name,
+      ordinal_position,
+      character_maximum_length,
+      data_type
+    FROM information_schema.columns isc
+    where table_schema = UPPER(:3)
+    AND table_name = UPPER(:4)
+    ORDER BY ordinal_position
+    ),
+
+  varchar_check AS (
+    SELECT listagg(
+          CASE WHEN T.character_maximum_length < s.character_maximum_length THEN s.column_name END
+          , ', ') WITHIN GROUP (ORDER BY s.ordinal_position) AS cols_with_varchar_issue
+    FROM target_data  t
+    LEFT JOIN SOURCE_data s
+    ON t.ordinal_position = s.ordinal_position AND t.column_name = s.column_name
+    WHERE t.ordinal_position>= s.ordinal_position),
+
+  type_check AS (
+    SELECT SUM(CASE WHEN s.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_source,
+          SUM(CASE WHEN t.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_target
+    FROM target_data  t
+    FULL OUTER JOIN SOURCE_data s
+    ON t.ordinal_position = s.ordinal_position AND T.data_type = s.data_type AND s.column_name = t.column_name
+    WHERE t.ordinal_position>= s.ordinal_position OR s.ordinal_position is null),
+
+  columns_to_add AS (
+    SELECT listagg(CASE WHEN t.ordinal_position IS NULL THEN s.column_definition end
+    , ', ') WITHIN GROUP (ORDER BY s.ordinal_position) AS cols_to_add
+    FROM source_data s
+    LEFT JOIN target_data t
+    ON t.ordinal_position = s.ordinal_position )
+
+  SELECT missing_in_source, missing_in_target, cols_with_varchar_issue, cols_to_add
+  FROM columns_to_add, type_check, varchar_check`;
+
+  var res = snowflake.createStatement({sqlText: column_check_stmt,
+                                       binds: [SOURCE_SCHEMA, SOURCE_TABLE,TARGET_SCHEMA, TARGET_TABLE]}
+                                      ).execute();
+  res.next();
+
+  missing_in_source = res.getColumnValue(1);
+  missing_in_target = res.getColumnValue(2);
+  cols_with_varchar_issue = res.getColumnValue(3);
+  cols_to_add = res.getColumnValue(4);
+
+  if (missing_in_target > 1) {
```

Comment:
This doesn't seem to align with the error message. Should this not be `missing_in_source > 0`?